resolver/resolvers.go

```diff
@@ -200,10 +212,162 @@ func createResolver(resolverURL, source string) (*Resolver, bool, error) {
 		}
 	}
 
+	// Resolve ip if was not specfied by the user
+	err = checkAndResolveServerAddressAndPort(newResolver)
+	if err != nil {
+		return nil, false, err
+	}
+
 	newResolver.Conn = resolverConnFactory(newResolver)
 	return newResolver, false, nil
 }
 
+func checkURLParameterValidity(scheme string, isHostnameDomain bool, query url.Values) error {
+	for key := range query {
+		switch key {
+		case parameterName,
+			parameterVerify,
+			parameterIP,
+			parameterBlockedIf,
+			parameterSearch,
+			parameterSearchOnly,
+			parameterPath:
+			// Known key, continue.
+		default:
+			// Unknown key, abort.
+			return fmt.Errorf(`unknown parameter "%s"`, key)
+		}
+	}
+
+	verifyDomain := query.Get(parameterVerify)
+	paramterServerIP := query.Get(parameterIP)
+
+	if scheme == ServerTypeDoT || scheme == ServerTypeDoH {
+
+		switch {
+		case isHostnameDomain && verifyDomain != "":
+			return fmt.Errorf("cannot have verify parameter with a domain as a hostname")
+		case !isHostnameDomain && verifyDomain == "":
+			return fmt.Errorf("verify paremeter must be set when using ip as domain")
+		case !isHostnameDomain && paramterServerIP != "":
+			return fmt.Errorf("cannot have ip parameter while domain is an ip")
```

Comment:
suggest: `cannot set the IP address via both the hostname in the URL and the ip parameter`